Packages/com.unity.inputsystem/InputSystem/Plugins/XR/TrackedPoseDriver.cs

```diff
@@ -581,18 +581,23 @@ protected virtual void SetLocalTransform(Vector3 newPosition, Quaternion newRota
             var positionValid = m_IgnoreTrackingState || (m_CurrentTrackingState & TrackingStates.Position) != 0;
             var rotationValid = m_IgnoreTrackingState || (m_CurrentTrackingState & TrackingStates.Rotation) != 0;
 
-            if (rotationValid &&
-                (m_TrackingType == TrackingType.RotationAndPosition ||
-                 m_TrackingType == TrackingType.RotationOnly))
+            if (rotationValid && positionValid && m_TrackingType == TrackingType.RotationAndPosition)
             {
-                transform.localRotation = newRotation;
+                transform.SetLocalPositionAndRotation(newPosition, newRotation);
             }
-
-            if (positionValid &&
-                (m_TrackingType == TrackingType.RotationAndPosition ||
-                 m_TrackingType == TrackingType.PositionOnly))
+            else
             {
-                transform.localPosition = newPosition;
+                if (rotationValid &&
+                m_TrackingType == TrackingType.RotationOnly)
```

Comment:
This won't be exactly the correct behavior we want if rotation is valid but position is not when tracking type is RotationAndPosition. I believe I have the correct code we want in another PR https://github.com/Unity-Technologies/InputSystem/pull/1716

I think we can close this PR in favor of that one if you would like.

Thanks again for the improvement suggestion (and for suggesting `Transform.SetLocalPositionAndRotation` itself).